File: C++/038. Count and Say.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    std::string countAndSay(int n) {
        std::vector<std::string> res {"", "1"};
        for (int i = 2; i <= n; i++) {
            std::string str = res[i - 1];
            res.emplace_back("");
            int num = 1, j = 1;
            for (; j < str.size(); j++) {
                if (str[j] == str[j - 1])
                    num++;
                else {
                    res[i] += std::to_string(num) + str[j - 1];
                    num = 1;
                }
            }
            res[i] += std::to_string(num) + str[j - 1];
        }
        return res[n];
    }
};

File: Go/038. Count and Say.go
package main

import "strconv"

func countAndSay(n int) string {
	res := make(map[int]string)
	res[1] = "1"
	for i := 2; i <= n; i++ {
		str := res[i - 1]
		res[i] = ""
		n, j := 1, 1
		for ; j < len(str); j++ {
			if str[j] == str[j - 1] {
				n++
			} else {
				res[i] += strconv.Itoa(n) + str[j - 1 : j]
				n = 1
			}
		}
		res[i] += strconv.Itoa(n) + str[j - 1 : j]
	}
	return res[n]
}